apis/core/v1alpha2/function_conversion.go

```diff
@@ -277,74 +307,92 @@ func (dst *Function) convertServingFrom(src *v1alpha1.Function) error {
 		dst.Spec.Serving.Template = src.Spec.Serving.Template
 	}
 
-	if src.Spec.Serving.OpenFuncAsync != nil {
-		dst.Spec.Serving.OpenFuncAsync = &OpenFuncAsyncRuntime{}
-		if src.Spec.Serving.OpenFuncAsync.Dapr != nil {
-			dst.Spec.Serving.OpenFuncAsync.Dapr = &Dapr{}
-			if src.Spec.Serving.OpenFuncAsync.Dapr.Annotations != nil {
-				dst.Spec.Serving.OpenFuncAsync.Dapr.Annotations = src.Spec.Serving.OpenFuncAsync.Dapr.Annotations
-			}
+	if src.Spec.Serving.Runtime == v1beta1.Async {
+		dst.Spec.Serving.OpenFuncAsync = &OpenFuncAsyncRuntime{
+			Dapr: &Dapr{},
+		}
+		dst.Spec.Serving.OpenFuncAsync.Dapr.Annotations = map[string]string{}
+		dst.Spec.Serving.OpenFuncAsync.Dapr.Components = map[string]*componentsv1alpha1.ComponentSpec{}
 
-			if src.Spec.Serving.OpenFuncAsync.Dapr.Components != nil {
-				dst.Spec.Serving.OpenFuncAsync.Dapr.Components = map[string]*componentsv1alpha1.ComponentSpec{}
-				for _, component := range src.Spec.Serving.OpenFuncAsync.Dapr.Components {
-					dst.Spec.Serving.OpenFuncAsync.Dapr.Components[component.Name] = &component.ComponentSpec
-				}
-			}
+		if src.Spec.Serving.ScaleOptions != nil && src.Spec.Serving.ScaleOptions.Keda != nil {
+			dst.Spec.Serving.OpenFuncAsync.Keda = &Keda{}
+		}
+	}
 
-			if src.Spec.Serving.OpenFuncAsync.Dapr.Inputs != nil {
-				for _, input := range src.Spec.Serving.OpenFuncAsync.Dapr.Inputs {
-					in := DaprIO{
-						Name:      fmt.Sprintf("%s-%s-%s", src.Namespace, src.Name, input.Name),
-						Component: input.Name,
-						Topic:     input.Topic,
-						Params:    input.Params,
-					}
-					dst.Spec.Serving.OpenFuncAsync.Dapr.Inputs = append(dst.Spec.Serving.OpenFuncAsync.Dapr.Inputs, &in)
-				}
-			}
+	if src.Spec.Serving.Bindings != nil {
+		for name, component := range src.Spec.Serving.Bindings {
+			dst.Spec.Serving.OpenFuncAsync.Dapr.Components[name] = component
```

Comment:
still component is a pointer, we cannot assign a pointer loop variable to some other variables. Consider using deepcopy